### ISSUE TITLE
cypress/no-async-before rule fails with "definition not found"

### ISSUE THREAD

**Issue by MikeMcC399** [cypress-io/eslint-plugin-cypress#190]
## Issue

- The rule [no-async-before](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/no-async-before.md) provided by PR https://github.com/cypress-io/eslint-plugin-cypress/pull/151 is not found when attempting to use it:

```text
  1:1  error  Definition for rule 'cypress/no-async-before' was not found  cypress/no-async-before
```

Additionally there are documentation errors and omissions:

1. The entry for this rule is missing in the [README > Rules](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/README.md#rules) section, although it is mentioned in the [README > Usage](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/README.md#usage) section.
2. The documentation heading for [no-async-before](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/no-async-before.md) "Prevent using async/await in Cypress test cases (no-async-tests)" is incorrect and is the same as [no-async-tests](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/no-async-test.md).

## Steps to reproduce

Ubuntu `22.04.4` LTS, Node.js `v22.12.2` LTS

Execute

```shell
mkdir no-async-before-test
cd no-async-before-test
git init
npm init -y
npm install eslint@8 eslint-plugin-cypress@latest
```

Create a `.eslintrc.json` file

```json
{
  "plugins": [
    "cypress"
  ],
    "rules": {
    "cypress/no-async-tests": "error",
    "cypress/no-async-before": "error"
  }
}
```

```shell
touch index.js
npx eslint .
```

## Logs

```text
$ npx eslint .

  1:1  error  Definition for rule 'cypress/no-async-before' was not found  cypress/no-async-before

✖ 1 problem (1 error, 0 warnings)
```

**Comment by MikeMcC399** [comment]
- PR #191 and [eslint-plugin-cypress/3.0.1](https://github.com/cypress-io/eslint-plugin-cypress/releases/tag/v3.0.1) resolve the issue cypress/no-async-before rule fails with "definition not found"
- The remaining documentation issues are transferred to the new issue https://github.com/cypress-io/eslint-plugin-cypress/issues/193
